fe2-android/app/src/main/java/com/github/dedis/student20_pop/model/network/method/message/ElectionVote.java

```diff
@@ -0,0 +1,100 @@
+package com.github.dedis.student20_pop.model.network.method.message;
+
+import com.github.dedis.student20_pop.utility.security.Hash;
+import com.google.gson.annotations.SerializedName;
+
+import java.util.List;
+
+public class ElectionVote  {
+
+    private String id; /** Id of the object ElectionVote : Hash(“Vote”||election_id||
+     || question_id||(vote_index(es)|write_in))
+     **/
+    @SerializedName(value = "question")
+    private String questionId; // id of the question
+    private List<Integer> vote; // list of indexes for the votes
+    @SerializedName(value = "write_in")
+    private Boolean writeIn; // enables to write in in ballot options
+
+    /**
+     * Constructor for a data Question, for the election setup
+     * @param questionId the Id of the question
+     * @param vote the list of indexes for the ballot options chose by the voter
+     * @param writeIn enables write in
+     * @param electionId Id of the election
+     */
+    public ElectionVote(
+            String questionId,
+            List<Integer> vote,
+            Boolean writeIn,
+            String electionId) {
+
+        this.questionId = questionId;
+        this.writeIn = writeIn;
+        this.vote = vote;
+        this.id = Hash.hash("Vote", electionId, questionId, vote.toString(), writeIn.toString());
```

Comment:
In the votes of [dataCastVote](https://github.com/dedis/student_21_pop/blob/master/protocol/query/method/message/data/dataCastVote.json)  there is `oneOf` vote and write_in, this means that the votes have either a write_in or a vote but not both

Reply:
+1